aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -357,6 +357,10 @@ Tensor dot_cuda(const Tensor& self, const Tensor& other) {
     incy = 1;
   }
 
+if (self._is_zerotensor() || other._is_zerotensor()) {
+  return at::_efficientzerotensor({}, self.options());
```

Comment:
Is it okay to just use `self.options()`? I see that `add_zerotensor` has some more logic to figure out the device, type promotion etc. Do we also want to do that here?

Reply:
If you look at line 368, you'll see that result is created with `self.options()`. The function called above just performs checks to validate that both inputs have same devices etc.

Reply:
I see, thanks!